### ISSUE TITLE
MD052/MD053: false positive on link-like text inside raw HTML tags (e.g. <code>, <var>), unlike backtick code spans

### ISSUE THREAD

**Issue by minseonkkim** [DavidAnson/markdownlint#2178]
### What did you do?

I wrote inline code using HTML tags instead of a backtick code span, so I could nest a `<var>` element for placeholder styling (something a plain backtick code span can't express):

```md
<code>Object.create(<var>proxy</var>)[foo]</code>
```

For comparison, the same content as a backtick code span does **not** trigger any warning:

```md
`Object.create(proxy)[foo]`
```

### What did you expect to happen?

I expected `[foo]` inside `<code>...</code>` to be treated the same way as `[foo]` inside a backtick code span, i.e. excluded from reference-link checking, since it's just code text that happens to contain a `(...)  [...]` sequence, not an actual Markdown link.

### What actually happened?

MD052 / MD053 flagged `[foo]` as an undefined shortcut reference link ("no reference definition"), because text inside raw HTML tags is not excluded from link-reference checking the way backtick code spans are.

### What messages or errors were there?

```
MD053/link-image-reference-definitions Link and image reference definitions should be needed [Unused link or image reference definition: "foo"]
```
(exact rule/message may vary depending on which of MD052/MD053 is enabled; both apply to this pattern)

### How can the issue be reproduced?

Minimal repro — save as `test.md` and lint it:

```md
<code>Object.create(<var>proxy</var>)[foo]</code>
```

Expected: no warning (same as the backtick-span version below).
Actual: MD052/MD053 warning about `[foo]`.

```md
`Object.create(proxy)[foo]`
```

This one (backtick code span) correctly produces no warning, showing the inconsistency.

### Real-world context

This came up in mdn/translated-content, where translated code samples use `<code>...<var>...</var>...</code>` per the project's style guide (needed for placeholder styling that plain backticks can't provide):

- https://github.com/mdn/translated-content/pull/28472
- Current file still working around it with a zero-width space (`&#8203;`) inserted between
  `)` and `[` purely to defeat the false positive:  https://github.com/mdn/translated-content/blob/main/files/ko/web/javascript/guide/meta_programming/index.md (search for `&#8203;`)

### Suggested fix

Extend the text-extraction logic already used to exclude backtick code spans from MD052/MD053 so it also excludes text inside `<code>`, `<kbd>`, `<pre>`, and their nested HTML children (e.g. `<var>`).

### What version were you using?

markdownlint-cli2 0.23.1 (using markdownlint core 0.41.1), as pinned in mdn/translated-content's package-lock.json

### What operating system were you using?

Windows

**Comment by DavidAnson** [comment]
Your premise that HTML code spans and Markdown code spans are equivalent is not correct. CommonMark allows Markdown syntax inside HTML content. The example below shows italics and a link inside an HTML code span. Therefore, reversed links are possible inside an HTML code span and therefore it is correct for this rule to report the issue.

https://dlaa.me/markdownlint/#%25m%23%20Issue%202178%0A%0A%3Ccode%3EMarkdown%20*allows*%20markup%20%5Blike%20links%5D(destination)%20in%20HTML.%3C%2Fcode%3E%0A

**Comment by DavidAnson** [comment]
Note: The title and text of the issue discuss MD052/MD053, but the minimal reproduction example provided produces MD011 which is what my comment refers to. The same principle should apply to other rules.
